Problem on AWICM3 installation on JUWELS

Hi all,

I have some problems when installing AWICM3 on JUWELS, which I think may be related to the netcdf-Fortran configuration.
I will be grateful if anyone can have a look at it. By the way, this issue cannot be reproduced in other HPCs, like ollie, mistral.

Best,
Jian

**System**
 - Supercomputer: JUWELS (Stages 2022, Python/3.9.6, netCDF-Fortran/4.5.3-serial)
 - ESM-Tools 6.1**
 
I used the command: **esm_master install-awicm3-deck**, and it shows errors after asking for permission of all the components: 
(I also alter different stages, python, netcdf versions, but the same errors remain)

-- setting default build type: Release
-- The C compiler identification is Intel 2021.4.0.20210910
-- The CXX compiler identification is Intel 2021.4.0.20210910
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /p/software/juwels/stages/2022/software/OpenMPI/4.1.2-intel-compilers-2021.4.0/bin/mpicc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /p/software/juwels/stages/2022/software/OpenMPI/4.1.2-intel-compilers-2021.4.0/bin/mpic++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The Fortran compiler identification is Intel 2021.4.0.20210910
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Check for working Fortran compiler: /p/software/juwels/stages/2022/software/OpenMPI/4.1.2-intel-compilers-2021.4.0/bin/mpifort - skipped
-- Checking whether /p/software/juwels/stages/2022/software/OpenMPI/4.1.2-intel-compilers-2021.4.0/bin/mpifort supports Fortran 90
-- Checking whether /p/software/juwels/stages/2022/software/OpenMPI/4.1.2-intel-compilers-2021.4.0/bin/mpifort supports Fortran 90 - yes
-- Looking for Fortran NCCRE
-- Looking for Fortran NCCRE - not found
-- Looking for nc__create
-- Looking for nc__create - not found
-- Configuring done
**CMake Error: The following variables are used in this project, but they are set to NOTFOUND.**
Please set them or make sure they are set and tested correctly in the CMake files:
/p/home/jusers/shi4/juwels/model_codes/awicm3-deck/oasis/lib/psmile/NETCDF_Fortran_INCLUDE_DIRECTORIES
   used as include directory in directory /p/home/jusers/shi4/juwels/model_codes/awicm3-deck/oasis/lib/psmile
/p/home/jusers/shi4/juwels/model_codes/awicm3-deck/oasis/lib/scrip/src/NETCDF_Fortran_INCLUDE_DIRECTORIES
   used as include directory in directory /p/home/jusers/shi4/juwels/model_codes/awicm3-deck/oasis/lib/scrip/src
NETCDF_Fortran_LIBRARIES
    linked by target "psmile" in directory /p/home/jusers/shi4/juwels/model_codes/awicm3-deck/oasis/lib/psmile

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
Traceback (most recent call last):
  File "/p/home/jusers/shi4/juwels/.local/bin/esm_master", line 33, in <module>
    sys.exit(load_entry_point('esm-tools', 'console_scripts', 'esm_master')())
  File "/p/home/jusers/shi4/juwels/esm_tools/src/esm_master/cli.py", line 103, in main
    main_flow(parsed_args, target)
  File "/p/home/jusers/shi4/juwels/esm_tools/src/esm_master/esm_master.py", line 106, in main_flow
    user_task.execute(ignore_errors)  # env)
  File "/p/home/jusers/shi4/juwels/esm_tools/src/esm_master/task.py", line 494, in execute
    subprocess.run(
  File "/p/software/juwels/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['./comp-oasis3mct-4.0-oifsdeck_script.sh']' returned non-zero exit status 1.


**Loaded modules**
![image](https://user-images.githubusercontent.com/18346694/156032415-50a63556-4f38-4c69-83c7-7a239530612c.png)

Hi all,
I get the same error when trying to install awiesm-2.1 on JUWELS

```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
/p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/lib/psmile/NETCDF_Fortran_INCLUDE_DIRECTORIES
   used as include directory in directory /p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/lib/psmile
/p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/lib/scrip/src/NETCDF_Fortran_INCLUDE_DIRECTORIES
   used as include directory in directory /p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/lib/scrip/src
NETCDF_Fortran_LIBRARIES
    linked by target "psmile" in directory /p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/lib/psmile

-- Configuring incomplete, errors occurred!
See also "/p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/build/CMakeFiles/CMakeOutput.log".
See also "/p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/build/CMakeFiles/CMakeError.log".
Traceback (most recent call last):
  File "/p/home/jusers/ackermann1/juwels/.local/bin/esm_master", line 33, in <module>
    sys.exit(load_entry_point('esm-tools', 'console_scripts', 'esm_master')())
  File "/p/home/jusers/ackermann1/juwels/esm_tools/src/esm_master/cli.py", line 103, in main
    main_flow(parsed_args, target)
  File "/p/home/jusers/ackermann1/juwels/esm_tools/src/esm_master/esm_master.py", line 106, in main_flow
    user_task.execute(ignore_errors)  # env)
  File "/p/home/jusers/ackermann1/juwels/esm_tools/src/esm_master/task.py", line 494, in execute
    subprocess.run(
  File "/p/software/juwels/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['./comp-oasis3mct-2.8-paleodyn_script.sh']' returned non-zero exit status 1.
```

 - loaded modules: 
 ```
Currently Loaded Modules:
  1) Stages/2022     (S)   5) StdEnv/2022            9) Tcl/8.6.11        13) libffi/.3.4.2   (H)  17) libyaml/.0.2.5         (H)  21) NASM/.2.15.05        (H)
  2) GCCcore/.11.2.0 (H)   6) bzip2/.1.0.8     (H)  10) SQLite/.3.36 (H)  14) OpenSSL/1.1          18) PostgreSQL/13.4             22) libjpeg-turbo/.2.1.1 (H)
  3) zlib/.1.2.11    (H)   7) ncurses/.6.2     (H)  11) XZ/.5.2.5    (H)  15) libxml2/.2.9.10 (H)  19) gflags/.2.2.2          (H)  23) Python/3.9.6
  4) binutils/.2.37  (H)   8) libreadline/.8.1 (H)  12) GMP/6.2.1         16) libxslt/.1.1.34 (H)  20) libspatialindex/.1.9.3 (H)

  Where:
   S:  Module is Sticky, requires --force to unload or purge
   H:             Hidden Module
```
 - esm_tools version 6.1

I still get the same error:
```
-- Check for working Fortran compiler: /p/software/juwels/stages/2022/software/OpenMPI/4.1.2-intel-compilers-2021.4.0/bin/mpifort - skipped
-- Checking whether /p/software/juwels/stages/2022/software/OpenMPI/4.1.2-intel-compilers-2021.4.0/bin/mpifort supports Fortran 90
-- Checking whether /p/software/juwels/stages/2022/software/OpenMPI/4.1.2-intel-compilers-2021.4.0/bin/mpifort supports Fortran 90 - yes
-- Looking for Fortran NCCRE
-- Looking for Fortran NCCRE - not found
-- Looking for nc__create
-- Looking for nc__create - not found
-- Configuring done
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
/p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/lib/psmile/NETCDF_Fortran_INCLUDE_DIRECTORIES
   used as include directory in directory /p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/lib/psmile
/p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/lib/scrip/src/NETCDF_Fortran_INCLUDE_DIRECTORIES
   used as include directory in directory /p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/lib/scrip/src
NETCDF_Fortran_LIBRARIES
    linked by target "psmile" in directory /p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/lib/psmile
```

I changed the default in `juwels.yaml` to `compiler_mpi: intel2022_ompi2022` and this to:
```
    awi_libs:
       choose_compiler_mpi:
          intel2022_ompi2022:
             add_export_vars:
                #IO_LIB_ROOT: /p/project/chhb19/streffing1/software/HPC_libraries/intel-2021.4.0_openmpi-4.1.2_20220129
                IO_LIB_ROOT: /p/project/chhb20/HPC_libraries/intel-2021.4.0_openmpi-4.1.2_20220129
```

Is `PROJECT_ID` supposed to be set automatically? 

In any case, I run:
```
export PROJECT_ID=chhb20
esm_master clean-awiesm-2.1
esm_master comp-awiesm-2.1
```

and get the same error as before:

```
-- Looking for Fortran NCCRE
-- Looking for Fortran NCCRE - not found
-- Looking for nc__create
-- Looking for nc__create - not found
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
/p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/lib/psmile/NETCDF_Fortran_INCLUDE_DIRECTORIES
   used as include directory in directory /p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/lib/psmile
/p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/lib/scrip/src/NETCDF_Fortran_INCLUDE_DIRECTORIES
   used as include directory in directory /p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/lib/scrip/src
NETCDF_Fortran_LIBRARIES
    linked by target "psmile" in directory /p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/lib/psmile

-- Configuring incomplete, errors occurred!
See also "/p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/build/CMakeFiles/CMakeOutput.log".
See also "/p/home/jusers/ackermann1/juwels/awiesm-2.1/oasis/build/CMakeFiles/CMakeError.log".
```

--- a/configs/machines/juwels.yaml
+++ b/configs/machines/juwels.yaml
@@ -127,6 +127,7 @@ choose_compiler_mpi:
         - "load imkl/2021.4.0"
         - "load Perl/5.34.0"
         - "load NCO/5.0.3"
+        - "load CDO/2.0.2"
       add_export_vars:
         FC: mpifort
         F77: mpifort

--- a/configs/machines/juwels.yaml
+++ b/configs/machines/juwels.yaml
@@ -321,6 +321,7 @@ choose_iolibraries:
             intel2022_psmpi2022:
                 add_export_vars:
                     IO_LIB_ROOT: /p/project/hirace/HPC_libraries/intel2021.4.0_parastation_5.5.0-1-mt_20220222
+                    PSMPIFLAGS: '"-lrt -lm -ldl"'
 
 # some yamls use computer.fc, etc to identify the compiler, so we need to add them
 fc: "$FC"

--- a/configs/machines/juwels.yaml
+++ b/configs/machines/juwels.yaml
@@ -126,6 +126,7 @@ choose_compiler_mpi:
         - "load Python/3.9.6"
         - "load imkl/2021.4.0"
         - "load Perl/5.34.0"
+        - "load NCO/5.0.3"
       add_export_vars:
         FC: mpifort
         F77: mpifort

--- a/configs/machines/juwels.yaml
+++ b/configs/machines/juwels.yaml
@@ -249,10 +249,11 @@ choose_iolibraries:
        add_module_actions:
          - "load netCDF netCDF-Fortran"
     awi_libs:
+       project: !ENV ${PROJECT_ID}
        choose_compiler_mpi:
           intel2022_ompi2022:
              add_export_vars:
-                IO_LIB_ROOT: /p/project/chhb19/streffing1/software/HPC_libraries/intel-2021.4.0_openmpi-4.1.2_20220129
+                IO_LIB_ROOT: /p/project/${project}/HPC_libraries/intel-2021.4.0_openmpi-4.1.2_20220129
                 LD_LIBRARY_PATH: $IO_LIB_ROOT/lib:$LD_LIBRARY_PATH
 
                 SZIPROOT: $IO_LIB_ROOT
@@ -267,10 +268,10 @@ choose_iolibraries:
                 NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
                 NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
                 OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
-                PATH:  $IO_LIB_ROOT/bin:/p/scratch/chhb19/hhb193/miniconda3/bin/:$PATH
+                PATH: $PATH:$IO_LIB_ROOT/bin #:/p/scratch/chhb19/hhb193/miniconda3/bin/:$PATH
           intel2020_ompi2020:
              add_export_vars:
-                IO_LIB_ROOT: /p/project/chhb19/streffing1/software/HPC_libraries/intel2020.2.254_OpenMPI_4.1.0rc1_20210920
+                IO_LIB_ROOT: /p/project/${project}/HPC_libraries/intel2020.2.254_OpenMPI_4.1.0rc1_20210920
                 LD_LIBRARY_PATH: $IO_LIB_ROOT/lib:$LD_LIBRARY_PATH
 
                 SZIPROOT: $IO_LIB_ROOT
@@ -285,10 +286,10 @@ choose_iolibraries:
                 NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
                 NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
                 OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
-                PATH:  $IO_LIB_ROOT/bin:/p/scratch/chhb19/hhb193/miniconda3/bin/:$PATH
+                PATH: $PATH:$IO_LIB_ROOT/bin #:/p/scratch/chhb19/hhb193/miniconda3/bin/:$PATH
           intel2020_psmpi2020:
              add_export_vars:
-                IO_LIB_ROOT: /p/project/chhb19/streffing1/software/HPC_libraries/intel-2020.2.254-GCC-9.3.0_parastation-5.4.7-1-mt_20220111
+                IO_LIB_ROOT: /p/project/${project}/HPC_libraries/intel-2020.2.254-GCC-9.3.0_parastation-5.4.7-1-mt_20220111
                 LD_LIBRARY_PATH: $IO_LIB_ROOT/lib:$LD_LIBRARY_PATH
 
                 SZIPROOT: $IO_LIB_ROOT
@@ -303,10 +304,10 @@ choose_iolibraries:
                 NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
                 NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
                 OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
-                PATH:  $IO_LIB_ROOT/bin:/p/scratch/chhb19/hhb193/miniconda3/bin/:$PATH
+                PATH: $PATH:$IO_LIB_ROOT/bin #:/p/scratch/chhb19/hhb193/miniconda3/bin/:$PATH
           intel_2019a_psmpi2019:
              add_export_vars:
-                IO_LIB_ROOT: /p/project/chhb19/streffing1/software/HPC_libraries/intel2019.3.199_parastation_5.4.4-1-mt_20210802
+                IO_LIB_ROOT: /p/project/${project}/HPC_libraries/intel2019.3.199_parastation_5.4.4-1-mt_20210802
                 LD_LIBRARY_PATH: $IO_LIB_ROOT/lib:$LD_LIBRARY_PATH
 
                 SZIPROOT: $IO_LIB_ROOT
@@ -321,7 +322,7 @@ choose_iolibraries:
                 NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
                 NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
                 OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
-                PATH:  $IO_LIB_ROOT/bin:/p/scratch/chhb19/hhb193/miniconda3/bin/:$PATH
+                PATH: $PATH:$IO_LIB_ROOT/bin #:/p/scratch/chhb19/hhb193/miniconda3/bin/:$PATH
     geomar_libs:
        choose_compiler_mpi:
           intel2019_impi2019:

--- a/configs/machines/juwels.yaml
+++ b/configs/machines/juwels.yaml
@@ -244,183 +244,84 @@ choose_compiler_mpi:
 
 choose_iolibraries:
 
+    # DEFAULT LIBS
     system_libs:
        # TODO: find the correct libraries and dependencies
        add_module_actions:
          - "load netCDF netCDF-Fortran"
+
+    # AWI LIBS
     awi_libs:
-       project: !ENV ${PROJECT_ID}
-       choose_compiler_mpi:
-          intel2022_ompi2022:
-             add_export_vars:
-                IO_LIB_ROOT: /p/project/${project}/HPC_libraries/intel-2021.4.0_openmpi-4.1.2_20220129
-                LD_LIBRARY_PATH: $IO_LIB_ROOT/lib:$LD_LIBRARY_PATH
+        add_export_vars:
+            PROJECT_ID: '$(for i in $(id -Gn $whoami);do echo "  - $i" ;done | grep chhb | head -1 | cut -c 5-10)'
+            IO_LIB_ROOT: "<WILL_BE_OVERWERITTEN>"
+            LD_LIBRARY_PATH: $IO_LIB_ROOT/lib:$LD_LIBRARY_PATH
 
-                SZIPROOT: $IO_LIB_ROOT
-                HDF5ROOT: $IO_LIB_ROOT
-                HDF5_ROOT: $HDF5ROOT
-                NETCDFROOT: $IO_LIB_ROOT
-                NETCDFFROOT: $IO_LIB_ROOT
-                ECCODESROOT: $IO_LIB_ROOT
+            SZIPROOT: $IO_LIB_ROOT
+            HDF5ROOT: $IO_LIB_ROOT
+            HDF5_ROOT: $HDF5ROOT
+            NETCDFROOT: $IO_LIB_ROOT
+            NETCDFFROOT: $IO_LIB_ROOT
+            ECCODESROOT: $IO_LIB_ROOT
 
-                HDF5_C_INCLUDE_DIRECTORIES: $HDF5_ROOT/include
-                NETCDF_Fortran_INCLUDE_DIRECTORIES: $NETCDFFROOT/include
-                NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
-                PATH: $PATH:$IO_LIB_ROOT/bin #:/p/scratch/chhb19/hhb193/miniconda3/bin/:$PATH
-          intel2020_ompi2020:
-             add_export_vars:
-                IO_LIB_ROOT: /p/project/${project}/HPC_libraries/intel2020.2.254_OpenMPI_4.1.0rc1_20210920
-                LD_LIBRARY_PATH: $IO_LIB_ROOT/lib:$LD_LIBRARY_PATH
+            HDF5_C_INCLUDE_DIRECTORIES: $HDF5_ROOT/include
+            NETCDF_Fortran_INCLUDE_DIRECTORIES: $NETCDFFROOT/include
+            NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
+            NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
+            OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
+            PATH: $PATH:$IO_LIB_ROOT/bin #:/p/scratch/chhb19/hhb193/miniconda3/bin/:$PATH
 
-                SZIPROOT: $IO_LIB_ROOT
-                HDF5ROOT: $IO_LIB_ROOT
-                HDF5_ROOT: $HDF5ROOT
-                NETCDFROOT: $IO_LIB_ROOT
-                NETCDFFROOT: $IO_LIB_ROOT
-                ECCODESROOT: $IO_LIB_ROOT
+        choose_compiler_mpi:
+            intel2022_ompi2022:
+                add_export_vars:
+                    IO_LIB_ROOT: /p/project/$PROJECT_ID/HPC_libraries/intel-2021.4.0_openmpi-4.1.2_20220129
+            intel2020_ompi2020:
+                add_export_vars:
+                    IO_LIB_ROOT: /p/project/$PROJECT_ID/HPC_libraries/intel2020.2.254_OpenMPI_4.1.0rc1_20210920
+            intel2020_psmpi2020:
+                add_export_vars:
+                    IO_LIB_ROOT: /p/project/$PROJECT_ID/HPC_libraries/intel-2020.2.254-GCC-9.3.0_parastation-5.4.7-1-mt_20220111
+            intel_2019a_psmpi2019:
+                add_export_vars:
+                    IO_LIB_ROOT: /p/project/$PROJECT_ID/HPC_libraries/intel2019.3.199_parastation_5.4.4-1-mt_20210802
 
-                HDF5_C_INCLUDE_DIRECTORIES: $HDF5_ROOT/include
-                NETCDF_Fortran_INCLUDE_DIRECTORIES: $NETCDFFROOT/include
-                NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
-                PATH: $PATH:$IO_LIB_ROOT/bin #:/p/scratch/chhb19/hhb193/miniconda3/bin/:$PATH
-          intel2020_psmpi2020:
-             add_export_vars:
-                IO_LIB_ROOT: /p/project/${project}/HPC_libraries/intel-2020.2.254-GCC-9.3.0_parastation-5.4.7-1-mt_20220111
-                LD_LIBRARY_PATH: $IO_LIB_ROOT/lib:$LD_LIBRARY_PATH
-
-                SZIPROOT: $IO_LIB_ROOT
-                HDF5ROOT: $IO_LIB_ROOT
-                HDF5_ROOT: $HDF5ROOT
-                NETCDFROOT: $IO_LIB_ROOT
-                NETCDFFROOT: $IO_LIB_ROOT
-                ECCODESROOT: $IO_LIB_ROOT
-
-                HDF5_C_INCLUDE_DIRECTORIES: $HDF5_ROOT/include
-                NETCDF_Fortran_INCLUDE_DIRECTORIES: $NETCDFFROOT/include
-                NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
-                PATH: $PATH:$IO_LIB_ROOT/bin #:/p/scratch/chhb19/hhb193/miniconda3/bin/:$PATH
-          intel_2019a_psmpi2019:
-             add_export_vars:
-                IO_LIB_ROOT: /p/project/${project}/HPC_libraries/intel2019.3.199_parastation_5.4.4-1-mt_20210802
-                LD_LIBRARY_PATH: $IO_LIB_ROOT/lib:$LD_LIBRARY_PATH
-
-                SZIPROOT: $IO_LIB_ROOT
-                HDF5ROOT: $IO_LIB_ROOT
-                HDF5_ROOT: $HDF5ROOT
-                NETCDFROOT: $IO_LIB_ROOT
-                NETCDFFROOT: $IO_LIB_ROOT
-                ECCODESROOT: $IO_LIB_ROOT
-
-                HDF5_C_INCLUDE_DIRECTORIES: $HDF5_ROOT/include
-                NETCDF_Fortran_INCLUDE_DIRECTORIES: $NETCDFFROOT/include
-                NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
-                PATH: $PATH:$IO_LIB_ROOT/bin #:/p/scratch/chhb19/hhb193/miniconda3/bin/:$PATH
+    # GEOMAR LIBS
     geomar_libs:
-       choose_compiler_mpi:
-          intel2019_impi2019:
-             add_export_vars:
-                IO_LIB_ROOT: /p/project/hirace/HPC_libraries/intel2019.3.199_impi2019.6.154_20200703
-                PATH: $IO_LIB_ROOT/bin:$PATH
-                LD_LIBRARY_PATH: $IO_LIB_ROOT/lib:$LD_LIBRARY_PATH
+        add_export_vars:
+            IO_LIB_ROOT: "<WILL_BE_OVERWERITTEN>"
+            PATH: $IO_LIB_ROOT/bin:$PATH
+            LD_LIBRARY_PATH: $IO_LIB_ROOT/lib:$LD_LIBRARY_PATH
 
-                SZIPROOT: $IO_LIB_ROOT
-                HDF5ROOT: $IO_LIB_ROOT
-                HDF5_ROOT: $HDF5ROOT
-                NETCDFROOT: $IO_LIB_ROOT
-                NETCDFFROOT: $IO_LIB_ROOT
-                ECCODESROOT: $IO_LIB_ROOT
+            SZIPROOT: $IO_LIB_ROOT
+            HDF5ROOT: $IO_LIB_ROOT
+            HDF5_ROOT: $HDF5ROOT
+            NETCDFROOT: $IO_LIB_ROOT
+            NETCDFFROOT: $IO_LIB_ROOT
+            ECCODESROOT: $IO_LIB_ROOT
 
-                HDF5_C_INCLUDE_DIRECTORIES: $HDF5_ROOT/include
-                NETCDF_Fortran_INCLUDE_DIRECTORIES: $NETCDFFROOT/include
-                NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
+            HDF5_C_INCLUDE_DIRECTORIES: $HDF5_ROOT/include
+            NETCDF_Fortran_INCLUDE_DIRECTORIES: $NETCDFFROOT/include
+            NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
+            NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
+            OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
 
-          intel2020_impi2020:
-             add_export_vars:
-                IO_LIB_ROOT: /p/project/hirace/HPC_libraries/intel2020.2.254_impi2019.8.254_20210427
-                PATH: $IO_LIB_ROOT/bin:$PATH
-                LD_LIBRARY_PATH: $IO_LIB_ROOT/lib:$LD_LIBRARY_PATH
+        choose_compiler_mpi:
+            intel2019_impi2019:
+                add_export_vars:
+                    IO_LIB_ROOT: /p/project/hirace/HPC_libraries/intel2019.3.199_impi2019.6.154_20200703
+            intel2020_impi2020:
+                add_export_vars:
+                    IO_LIB_ROOT: /p/project/hirace/HPC_libraries/intel2020.2.254_impi2019.8.254_20210427
+            intel_2019a_psmpi2019:
+                add_export_vars:
+                    IO_LIB_ROOT: /p/project/hirace/HPC_libraries/intel2019.5.281_parastation_5.4.4-1-mt_20201113
+            intel2020_psmpi2020:
+                add_export_vars:
+                    IO_LIB_ROOT: /p/project/hirace/HPC_libraries/intel2020.2.254_parastation_5.4.7-1_20210427
+            intel2022_psmpi2022:
+                add_export_vars:
+                    IO_LIB_ROOT: /p/project/hirace/HPC_libraries/intel2021.4.0_parastation_5.5.0-1-mt_20220222
 
-                SZIPROOT: $IO_LIB_ROOT
-                HDF5ROOT: $IO_LIB_ROOT
-                HDF5_ROOT: $HDF5ROOT
-                NETCDFROOT: $IO_LIB_ROOT
-                NETCDFFROOT: $IO_LIB_ROOT
-                ECCODESROOT: $IO_LIB_ROOT
-
-                HDF5_C_INCLUDE_DIRECTORIES: $HDF5_ROOT/include
-                NETCDF_Fortran_INCLUDE_DIRECTORIES: $NETCDFFROOT/include
-                NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
-
-          intel_2019a_psmpi2019:
-             add_export_vars:
-                IO_LIB_ROOT: /p/project/hirace/HPC_libraries/intel2019.5.281_parastation_5.4.4-1-mt_20201113
-                PATH: $IO_LIB_ROOT/bin:$PATH
-                LD_LIBRARY_PATH: $IO_LIB_ROOT/lib:$LD_LIBRARY_PATH
-
-                SZIPROOT: $IO_LIB_ROOT
-                HDF5ROOT: $IO_LIB_ROOT
-                HDF5_ROOT: $HDF5ROOT
-                NETCDFROOT: $IO_LIB_ROOT
-                NETCDFFROOT: $IO_LIB_ROOT
-                ECCODESROOT: $IO_LIB_ROOT
-
-                HDF5_C_INCLUDE_DIRECTORIES: $HDF5_ROOT/include
-                NETCDF_Fortran_INCLUDE_DIRECTORIES: $NETCDFFROOT/include
-                NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
-
-          intel2020_psmpi2020:
-             add_export_vars:
-                IO_LIB_ROOT: /p/project/hirace/HPC_libraries/intel2020.2.254_parastation_5.4.7-1_20210427
-                PATH: $IO_LIB_ROOT/bin:$PATH
-                LD_LIBRARY_PATH: $IO_LIB_ROOT/lib:$LD_LIBRARY_PATH
-
-                SZIPROOT: $IO_LIB_ROOT
-                HDF5ROOT: $IO_LIB_ROOT
-                HDF5_ROOT: $HDF5ROOT
-                NETCDFROOT: $IO_LIB_ROOT
-                NETCDFFROOT: $IO_LIB_ROOT
-                ECCODESROOT: $IO_LIB_ROOT
-
-                HDF5_C_INCLUDE_DIRECTORIES: $HDF5_ROOT/include
-                NETCDF_Fortran_INCLUDE_DIRECTORIES: $NETCDFFROOT/include
-                NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
-
-          intel2022_psmpi2022:
-             add_export_vars:
-                IO_LIB_ROOT: /p/project/hirace/HPC_libraries/intel2021.4.0_parastation_5.5.0-1-mt_20220222 
-                PATH: $IO_LIB_ROOT/bin:$PATH
-                LD_LIBRARY_PATH: $IO_LIB_ROOT/lib:$LD_LIBRARY_PATH
-                PSMPIFLAGS: '"-lrt -lm -ldl"'
-
-                SZIPROOT: $IO_LIB_ROOT
-                HDF5ROOT: $IO_LIB_ROOT
-                HDF5_ROOT: $HDF5ROOT
-                NETCDFROOT: $IO_LIB_ROOT
-                NETCDFFROOT: $IO_LIB_ROOT
-                ECCODESROOT: $IO_LIB_ROOT
-
-                HDF5_C_INCLUDE_DIRECTORIES: $HDF5_ROOT/include
-                NETCDF_Fortran_INCLUDE_DIRECTORIES: $NETCDFFROOT/include
-                NETCDF_C_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                NETCDF_CXX_INCLUDE_DIRECTORIES: $NETCDFROOT/include
-                OASIS3MCT_FC_LIB: '"-L$NETCDFFROOT/lib -lnetcdff"'
-       #add_export_vars:
 # some yamls use computer.fc, etc to identify the compiler, so we need to add them
 fc: "$FC"
 cc: "$CC"

--- a/docs/ten_steps.rst
+++ b/docs/ten_steps.rst
@@ -7,10 +7,7 @@ Ten Steps to a Running Model
     $ export PATH=$PATH:~/.local/bin
     $ export LC_ALL=en_US.UTF-8
     $ export LANG=en_US.UTF-8
-    
-   If you are installing esm_tools on the Juwels HPC system, also add the following line to to one of your login or profile files, i.e. ``~/.bash_profile``, ``~/.bashrc``, ``~/.profile``, etc.::
 
-    $ export PROJECT_ID=$(for i in $(id -Gn $whoami);do echo "  - $i" ;done | grep chhb | head -1 | cut -c 5-10)
 
 2. Make sure you have a GitHub account and check our GitHub repository (``https://github.com/esm-tools``).
 

--- a/docs/ten_steps.rst
+++ b/docs/ten_steps.rst
@@ -7,7 +7,10 @@ Ten Steps to a Running Model
     $ export PATH=$PATH:~/.local/bin
     $ export LC_ALL=en_US.UTF-8
     $ export LANG=en_US.UTF-8
+    
+   If you are installing esm_tools on the Juwels HPC system, also add the following line to to one of your login or profile files, i.e. ``~/.bash_profile``, ``~/.bashrc``, ``~/.profile``, etc.::
 
+    $ export PROJECT_ID=$(for i in $(id -Gn $whoami);do echo "  - $i" ;done | grep chhb | head -1 | cut -c 5-10)
 
 2. Make sure you have a GitHub account and check our GitHub repository (``https://github.com/esm-tools``).
 

--- a/setup.cfg
+++ b/setup.cfg
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 6.3.2
+current_version = 6.3.3
 commit = True
 tag = True
 

--- a/setup.py
+++ b/setup.py
@@ -92,6 +92,6 @@ setup(
     test_suite="tests",
     tests_require=test_requirements,
     url="https://github.com/esm-tools/esm_tools",
-    version="6.3.2",
+    version="6.3.3",
     zip_safe=False,
 )

--- a/src/esm_archiving/__init__.py
+++ b/src/esm_archiving/__init__.py
@@ -4,7 +4,7 @@
 
 __author__ = """Paul Gierz"""
 __email__ = "pgierz@awi.de"
-__version__ = "6.3.2"
+__version__ = "6.3.3"
 
 from .esm_archiving import (
     archive_mistral,

--- a/src/esm_calendar/__init__.py
+++ b/src/esm_calendar/__init__.py
@@ -2,6 +2,6 @@
 
 __author__ = """Dirk Barbi"""
 __email__ = "dirk.barbi@awi.de"
-__version__ = "6.3.2"
+__version__ = "6.3.3"
 
 from .esm_calendar import *

--- a/src/esm_cleanup/__init__.py
+++ b/src/esm_cleanup/__init__.py
@@ -2,4 +2,4 @@
 
 __author__ = """Dirk Barbi"""
 __email__ = "dirk.barbi@awi.de"
-__version__ = "6.3.2"
+__version__ = "6.3.3"

--- a/src/esm_database/__init__.py
+++ b/src/esm_database/__init__.py
@@ -2,4 +2,4 @@
 
 __author__ = """Dirk Barbi"""
 __email__ = "dirk.barbi@awi.de"
-__version__ = "6.3.2"
+__version__ = "6.3.3"

--- a/src/esm_environment/__init__.py
+++ b/src/esm_environment/__init__.py
@@ -2,6 +2,6 @@
 
 __author__ = """Dirk Barbi"""
 __email__ = "dirk.barbi@awi.de"
-__version__ = "6.3.2"
+__version__ = "6.3.3"
 
 from .esm_environment import *

--- a/src/esm_master/__init__.py
+++ b/src/esm_master/__init__.py
@@ -2,7 +2,7 @@
 
 __author__ = """Dirk Barbi"""
 __email__ = "dirk.barbi@awi.de"
-__version__ = "6.3.2"
+__version__ = "6.3.3"
 
 
 from . import database

--- a/src/esm_motd/__init__.py
+++ b/src/esm_motd/__init__.py
@@ -2,6 +2,6 @@
 
 __author__ = """Dirk Barbi"""
 __email__ = "dirk.barbi@awi.de"
-__version__ = "6.3.2"
+__version__ = "6.3.3"
 
 from .esm_motd import *

--- a/src/esm_parser/__init__.py
+++ b/src/esm_parser/__init__.py
@@ -2,7 +2,7 @@
 
 __author__ = """Dirk Barbi"""
 __email__ = "dirk.barbi@awi.de"
-__version__ = "6.3.2"
+__version__ = "6.3.3"
 
 
 from .yaml_to_dict import yaml_file_to_dict

--- a/src/esm_plugin_manager/__init__.py
+++ b/src/esm_plugin_manager/__init__.py
@@ -2,6 +2,6 @@
 
 __author__ = """Dirk Barbi, Paul Gierz, Sebastian Wahl"""
 __email__ = "dirk.barbi@awi.de"
-__version__ = "6.3.2"
+__version__ = "6.3.3"
 
 from .esm_plugin_manager import *

--- a/src/esm_profile/__init__.py
+++ b/src/esm_profile/__init__.py
@@ -2,6 +2,6 @@
 
 __author__ = """Dirk Barbi"""
 __email__ = "dirk.barbi@awi.de"
-__version__ = "6.3.2"
+__version__ = "6.3.3"
 
 from .esm_profile import *

--- a/src/esm_rcfile/__init__.py
+++ b/src/esm_rcfile/__init__.py
@@ -2,6 +2,6 @@
 
 __author__ = """Dirk Barbi"""
 __email__ = "dirk.barbi@awi.de"
-__version__ = "6.3.2"
+__version__ = "6.3.3"
 
 from .esm_rcfile import *

--- a/src/esm_runscripts/__init__.py
+++ b/src/esm_runscripts/__init__.py
@@ -2,7 +2,7 @@
 
 __author__ = """Dirk Barbi"""
 __email__ = "dirk.barbi@awi.de"
-__version__ = "6.3.2"
+__version__ = "6.3.3"
 
 from .sim_objects import *
 from .batch_system import *

--- a/src/esm_runscripts/oasis.py
+++ b/src/esm_runscripts/oasis.py
@@ -464,10 +464,10 @@ class oasis:
             thesefiles = glob.glob(field + "_" + exe + "_*.nc")
             print(thesefiles, flush=True)
             for thisfile in thesefiles:
-                print("cdo showtime " + thisfile + " 2>/dev/null | wc -w", flush=True)
+                print("cdo showtime " + thisfile + " 2>/dev/null | head -n 1 | wc -w", flush=True)
                 lasttimestep = (
                     subprocess.check_output(
-                        "cdo showtime " + thisfile + " 2>/dev/null | wc -w", shell=True
+                        "cdo showtime " + thisfile + " 2>/dev/null | head -n 1 | wc -w", shell=True
                     )
                     .decode("utf-8")
                     .rstrip()

--- a/src/esm_tests/__init__.py
+++ b/src/esm_tests/__init__.py
@@ -2,7 +2,7 @@
 
 __author__ = """Miguel Andres-Martinez"""
 __email__ = "miguel.andres-martinez@awi.de"
-__version__ = "6.3.2"
+__version__ = "6.3.3"
 
 from .initialization import *
 from .tests import *

--- a/src/esm_tools/__init__.py
+++ b/src/esm_tools/__init__.py
@@ -23,7 +23,7 @@ so it's just the dictionary representation of the YAML.
 
 __author__ = """Dirk Barbi, Paul Gierz"""
 __email__ = "dirk.barbi@awi.de"
-__version__ = "6.3.2"
+__version__ = "6.3.3"
 
 import functools
 import inspect

--- a/src/esm_utilities/__init__.py
+++ b/src/esm_utilities/__init__.py
@@ -2,4 +2,4 @@
 
 __author__ = """Paul Gierz"""
 __email__ = "pgierz@awi.de"
-__version__ = "6.3.2"
+__version__ = "6.3.3"

